### PR TITLE
Java release workflow with maven and github packages

### PR DESCRIPTION
**Pull request by kompotkot**
Docs: https://docs.github.com/en/actions/guides/publishing-java-packages-with-maven#publishing-packages-to-github-packages

### CODE REVIEW

**Comment by kompotkot** [comment]
@bugout-dev check
- [ ] Create free account at https://github.com/features/packages
- [ ] Add repo secrets `GITHUB_RELEASE_TOKEN` with scopes: "read access for the contents scope and write access for the packages scope"

File: .github/workflows/release-java.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-java@v2
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Publish package
+        run: mvn --batch-mode deploy
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_RELEASE_TOKEN }}
```
Comment (zomglings) [inline .github/workflows/release-java.yml:27]: use `secrets.GITHUB_TOKEN`. This is a special secret that gets generated dynamically for each GitHub Action workflow. More info on the `GITHUB_TOKEN` docs - https://docs.github.com/en/actions/reference/authentication-in-a-workflow

Comment (kompotkot) [reply .github/workflows/release-java.yml]: Done

**Review by zomglings — APPROVED** [review]
lgtm

**Pull request merged**